flink-java/src/main/java/org/apache/flink/api/java/utils/RequiredParameter.java

```diff
@@ -0,0 +1,149 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Facility to manage required parameters in user defined functions.
+ */
+public class RequiredParameter {
+
+	private static final String HELP_TEXT_PARAM_DELIMITER = "\t";
+	private static final String HELP_TEXT_LINE_DELIMITER = "\n";
+
+	private HashMap<String, Option> data;
+
+	public RequiredParameter() {
+		this.data = new HashMap<>();
+	}
+
+	public void add(Option option) throws RequiredParameterException {
+		if (!this.data.containsKey(option.getName())) {
+			this.data.put(option.getName(), option);
+		} else {
+			throw new RequiredParameterException("Option with passed key already exists. " +  option.getName());
+		}
+	}
+
+	/**
+	 * Check if all parameters defined as required have been supplied.
+	 *
+	 * @param parameterTool - parameters supplied by user.
+	 */
+	public void check(ParameterTool parameterTool) throws RequiredParameterException {
+		for (Option o : data.values()) {
+			String key = o.getName();
+			String shortKey = o.getAlt();
+
+			boolean longKeyUndefined = keyIsUndefined(key, parameterTool.data);
+			boolean shortKeyUndefined = keyIsUndefined(shortKey, parameterTool.data);
+			// check if the ParameterTool contains a value for either the full or short key (if short key is defined)
+			if (longKeyUndefined && shortKeyUndefined) {
+				throw new RequiredParameterException("Required parameter " + key + " is not present.");
+			}
+
+			// check if the ParameterTool does not contain values for both full and short key (if short key is defined)
+			if (parameterTool.data.containsKey(key) && parameterTool.data.containsKey(shortKey)) {
+				throw new RequiredParameterException("Required parameter " + key +
+						" is defined twice: on short and long version.");
+			}
+
+			if (!shortKeyUndefined) {
+				parameterTool.data.put(key, parameterTool.data.get(shortKey));
+				parameterTool.data.remove(shortKey);
+			}
+
+			// key is defined and has value, now check if it adheres to the type specified.
+			if (o.hasType() && !o.isCastableToDefinedType(parameterTool.data.get(key))) {
+				throw new RequiredParameterException("Parameter " + key + " cannot be cast to the specified type.");
+			}
+		}
+	}
+
+	/**
+	 * Check if all parameters defined as required have been supplied. If not use the default values
+	 * which have been supplied. If no default value is supplied for a missing parameter, an exception is thrown.
+	 *
+	 * @param parameterTool - parameters supplied by the user.
+	 */
+	public void checkAndPopulate(ParameterTool parameterTool) throws RequiredParameterException {
+		for (Option o : data.values()) {
```

Comment:
Shouldn't `checkAndPopulate` do what `check` is doing and a bit more (populating)?
Actually, I am a bit confused about the purpose of having two separate methods.
Is it possible to merge both into a single method and call it for example `applyTo`?